app/assets/javascripts/admin/order_cycles/controllers/incoming_controller.js.coffee

```diff
@@ -28,4 +28,3 @@ angular.module('admin.orderCycles').controller 'AdminOrderCycleIncomingCtrl', ($
   $scope.selectAllVariants = (exchange, selected) ->
     $scope.loadAllExchangeProducts(exchange).then ->
       $scope.setExchangeVariants(exchange, $scope.suppliedVariants(exchange.enterprise_id), selected)
-      $scope.$apply()
```

Comment:
was this causing the console error? this is needed for some cases... the case is:
- on the incoming exchanges when you have only a few pages loaded and you "select all", it will load all variants first and then select them all. in this case, this apply() is making the "select all" select box become selected, otherwise it wont re-render after all variants are loaded and selected and it will remain unchecked (which is incorrect).

maybe we need to run this apply only in this case, which is the case where loadAllExchangeProducts actually loads something. I'd guess the console error is for the case where loadAllExchangeProducts is not loading anything (all products are already loaded)

Reply:
Maybe I'll just revert it and open a tech-debt issue for now :+1:

Reply:
Issue: #4676